### ISSUE TITLE
Koa-Shopify-Auth  - App getting rejected by shopify due to outh redirection issue.

### ISSUE THREAD

**Issue by gauravsharma1996** [Shopify/quilt#1397]
I am using the latest koa-shopify-auth library version(3.1.59) for my embedded Shopify app authentication, I have set all the cookies as same site none. But it is getting rejected by Shopify.

This is the issue the Shopify have mentioned.
![Screenshot from 2020-04-22 13-32-21 (1)](https://user-images.githubusercontent.com/18066488/79961329-2f392a80-84a4-11ea-8ac8-c7a7e05f4359.jpg)

" The app must install successfully. App not requesting access to shop immediately after clicking "Add App". Expected OAuth to be initiated during install or reinstall at https://appstoretest5.myshopify.com/admin/oauth/request_grant but was directed to https://redirect_url. ".

**Comment by mazz-seven** [comment]
Hey,
Did you find a solution to this? I'm having the same problem

**Comment by gauravsharma1996** [comment]
> Hey,
> Did you find a solution to this? I'm having the same problem

No, I haven't found the solution for this. But in my case, it is working fine on the chrome browser with the latest update. I am still trying to contact the Shopify team regarding the same.

**Comment by mazz-seven** [comment]
My problem was, using the koa-shopify-auth and setting the App URL to `https://{HOST}/` clicking the Add App button is redirecting to `/auth` by the verifyRequest middleware. Causing the automatic checks of Shopify to reject my app. 

My solution is to catch the request to "/" before  verifyRequest and check if the app is installed or not then redirect to admin/oauth/authorize?... immediately.

Is that make sense?

**Comment by michenly** [comment]
cc @katiedavis

**Comment by beatlecz** [comment]
> My problem was, using the koa-shopify-auth and setting the App URL to `https://{HOST}/` clicking the Add App button is redirecting to `/auth` by the verifyRequest middleware. Causing the automatic checks of Shopify to reject my app.
> 
> My solution is to catch the request to "/" before verifyRequest and check if the app is installed or not then redirect to admin/oauth/authorize?... immediately.
> 
> Is that make sense?

Could you share an example how you've solved this issue? It will be very helpful for me too.
